test/server/proxy-test.js

```diff
@@ -2606,5 +2622,49 @@ describe('Proxy', () => {
                 testRedirectRequestStatusCode(202, false)
             ]);
         });
+
+        it('Should add the trailing slash to location header if url consist of origin (GH-1426)', () => {
+            proxy.openSession('http://127.0.0.1:2000/', session);
+
+            function testRedirectRequest (redirectLocation, trailingSlashIsNeeded) {
+                const encodedUrl = encodeURIComponent(redirectLocation);
+                const options    = {
+                    url: urlUtils.getProxyUrl('http://127.0.0.1:2000/location-header-with-trailing-slash/' + encodedUrl, {
+                        proxyHostname: '127.0.0.1',
+                        proxyPort:     1836,
+                        sessionId:     session.id,
+                        resourceType:  urlUtils.getResourceTypeString({ isIframe: true })
+                    }),
+
+                    resolveWithFullResponse: true,
+                    followRedirect:          false,
+                    simple:                  false,
+                };
+
+                return request(options)
+                    .then(res => {
+                        const parsedUrl = urlUtils.parseProxyUrl(res.headers['location']);
+
+                        expect(parsedUrl.destUrl).eql(redirectLocation + (trailingSlashIsNeeded ? '/' : ''));
+                    });
+            }
+
+            return Promise.all([
+                testRedirectRequest('http://example.com', true),
```

Comment:
Reduce testcases count according the comment above.

Reply:
Done.